src/main/java/racingcar/RacingCar.java

```diff
@@ -0,0 +1,208 @@
+     * @see #getWinner()
+     * @see #showWinnerList(Vector)
+     */
+    public void run() {
+        int round;
+        Vector<User> winnerList = new Vector<>();
+
+        System.out.println("경주할 자동차 이름을 입력하세요.(이름은 쉼표(,) 기준으로 구분)");
+        String inputName = Console.readLine();
+        inputName = inputName.replaceAll(" ", "");
+        String usernameArray[] = inputName.split(",");
+        setUserList(usernameArray);
+
+        System.out.println("시도할 횟수는 몇 회인가요?");
+        try {
+            String inputRound = Console.readLine();
+            inputRound = inputRound.replaceAll(" ", "");
+            round = Integer.valueOf(inputRound);
+            checkRoundError(round);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException();
+        }
+
+        startRace(round);
+        winnerList = getWinner();
+        showWinnerList(winnerList);
+    }
+
+    /**
+     * 입력된 값이 유효한지 체크 <br>
+     * 입력된 이름으로 각각 참가자의 정보를 저장하는 User 객체를 할당하여 벡터에 추가
+     *
+     * @param usernameArray 유저가 입력한 자동차 이름 목록
+     * @throws IllegalArgumentException 이름이 입력 되지 않았음
+     * @see #run()
+     * @see #checkUsernameError(String)
+     */
+    private void setUserList(String usernameArray[]) {
+        if (usernameArray.length == 0) {
+            throw new IllegalArgumentException();
+        }
+
+        for (int i = 0; i < usernameArray.length; i++) {
+            checkUsernameError(usernameArray[i]);
+            userList.add(new User(usernameArray[i]));
+        }
+    }
+
+    /**
+     * 유저가 입력한 자동차 이름이 유효한 값인지 체크
+     *
+     * @param username 자동차 이름
+     * @throws IllegalArgumentException 입력된 이름이 빈 칸임
+     * @throws IllegalArgumentException 입력된 이름이 6자 이상임
+     * @see #setUserList(String[])
+     */
+    private void checkUsernameError(String username) {
+        if (username.isEmpty()) {
+            throw new IllegalArgumentException();
+        } else if (username.length() > 5) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    /**
+     * 유저가 입력한 경주를 시도할 횟수가 유효한 값인지 체크
+     *
+     * @param round 경주를 시도할 횟수
+     * @throws IllegalArgumentException 입력된 횟수가 음수임
+     * @see #run()
+     */
+    private void checkRoundError(int round) {
+        if (round < 0) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    /**
+     * 입력 받은 값 만큼 경주를 실행
+     *
+     * @param round 경주를 시도할 횟수
+     * @see #run()
+     * @see #runRaceForRound(int)
+     */
+    private void startRace(int round) {
+        System.out.println("\n실행 결과");
+        for (int i = 0; i < round; i++) {
+            runRaceForRound(i);
+            System.out.println();
+        }
+    }
+
+    /**
+     * 각 참가자의 라운드 결과를 저장, 출력
+     *
+     * @param roundIndex 현재 진행하고 있는 라운드
+     * @see #startRace(int)
+     * @see #goOrStay(User)
+     */
+    private void runRaceForRound(int roundIndex) {
+        Iterator<User> iterator = userList.iterator();
+        while (iterator.hasNext()) {
```

Comment:
for문을 사용하면 좀 더 가독성에 도움이 될 것 같습니다.
```suggestion
        for (User user : userList)) {
```